### PR TITLE
CEDS-2564 - convert view specs to use validating MessagesAPI

### PR DESCRIPTION
**Pull request by SteveSugden**
Draft PR to get feedback on direction of travel

### CODE REVIEW

File: test/views/declaration/AdditionalActorsRemoveViewSpec.scala [hunk 1]
```diff
@@ -70,7 +67,7 @@ class AdditionalActorsRemoveViewSpec extends UnitViewSpec with ExportsTestData w
         val view = createView()
 
         view.select("dl>div:nth-child(1)>dt").text() mustBe messages("declaration.additionalActors.table.party")
-        view.select("dl>div:nth-child(1)>dd").text() mustBe "declaration.partyType.MF"
+        view.select("dl>div:nth-child(1)>dd").text() mustBe messages("declaration.partyType.MF")
```
Comment (MaciejRewera) [inline test/views/declaration/AdditionalActorsRemoveViewSpec.scala:70]: Would matcher `containMessage` be more suitable?

File: app/views/declaration/additional_fiscal_references.scala.html [hunk 2]
```diff
@@ -82,7 +82,7 @@
 @inputFields = {
     @components.fields.field_accessible_autocomplete(
         field = form("country"),
-        label = messages("declaration.additionalFiscalReferences.country"),
+        label = "declaration.additionalFiscalReferences.country",
```
Comment (MaciejRewera) [inline app/views/declaration/additional_fiscal_references.scala.html:85]: IMHO it is better for the label to consume the final message, not the key. This way you have more loose coupling between `field_accessible_autocomplete` and its clients.

Comment (PatrykRudnicki) [reply app/views/declaration/additional_fiscal_references.scala.html:85]: But this is how it was implemented before.

But the new specs will recognise if you use `messages(messages("messagesKey"))` which is a reason of this change

File: test/views/declaration/AdditionalActorsSummaryViewSpec.scala [hunk 3]
```diff
@@ -69,23 +66,23 @@ class AdditionalActorsSummaryViewSpec extends UnitViewSpec with ExportsTestData
...
+        view.getElementsByTag("h1").text() mustBe messages("declaration.additionalActors.table.multiple.heading", "0")
       }
 
       "display page title for multiple items" in {
         createView(actors = Seq.empty).getElementsByTag("h1").text() mustBe messages("declaration.additionalActors.table.multiple.heading", "0")
       }
 
       "display section header" in {
-        view.getElementById("section-header").text() must include("declaration.summary.parties.header")
+        view.getElementById("section-header").text() must include(messages("declaration.summary.parties.header"))
```
Comment (MaciejRewera) [inline test/views/declaration/AdditionalActorsSummaryViewSpec.scala:77]: `containMessage` matcher seems like a way to go there.

File: test/views/declaration/spec/UnitViewSpec2.scala [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import org.jsoup.nodes.Document
+import org.scalatest.Assertion
+import org.scalatest.matchers.{BeMatcher, MatchResult}
+import play.api.i18n.{Lang, Messages, MessagesApi}
+import play.api.mvc.{AnyContent, Request}
+import play.api.test.{FakeRequest, Helpers}
+import services.cache.ExportsTestData
+import unit.base.{JourneyTypeTestRunner, UnitSpec}
+
+class UnitViewSpec2 extends UnitSpec with ViewMatchers with JourneyTypeTestRunner {
```
Comment (MaciejRewera) [inline test/views/declaration/spec/UnitViewSpec2.scala:29]: Nit: Isn't `JourneyTypeTestRunner` part of `UnitSpec` ?

File: app/views/declaration/additional_fiscal_references.scala.html (hunk 2 above)
Comment (MaciejRewera) [reply app/views/declaration/additional_fiscal_references.scala.html:85]: Alright, didn't realise that. No problem then (at least for now).

File: app/views/components/fields/field_accessible_autocomplete_with_header.scala.html [hunk 5]
```diff
@@ -47,7 +47,7 @@
 
     @if(isPageHeading) {<h1 class="govuk-label-wrapper">}
     <label id="@(fieldNameWithoutBrackets)-label" for="@fieldNameWithoutBrackets" class="govuk-label@if(labelClass.nonEmpty){ @labelClass}" >
-        @messages(label)
+        @if(label.nonEmpty) { @messages(label) }
```
Comment (SteveSugden) [inline app/views/components/fields/field_accessible_autocomplete_with_header.scala.html:50]: This change was added to fix country_of_routing change which is (I believe) incorrectly using an empty <label>

File: app/views/declaration/exporter_details.scala.html [hunk 6]
```diff
@@ -103,7 +103,7 @@
         @govukFieldset(Fieldset(
             attributes = Map("id" -> "details"),
                 legend = Some(Legend(
-                    content = Text(messages("")),
+                    content = Text(""),
```
Comment (SteveSugden) [inline app/views/declaration/exporter_details.scala.html:106]: Also looks wrong - label with no text

Comment (PatrykRudnicki) [reply app/views/declaration/exporter_details.scala.html:106]: In line 110 you have `exportsName`

You can move exportsName content to this legend content and remove exportsName.

With this approach you get rid of empty message here

Comment (SteveSugden) [reply app/views/declaration/exporter_details.scala.html:106]: Should I fix this bug as part of this ticket?  There was some other place with an empty label as well.

Comment (PatrykRudnicki) [reply app/views/declaration/exporter_details.scala.html:106]: I think it's not necessary, up to you :)

Comment (SteveSugden) [reply app/views/declaration/exporter_details.scala.html:106]: I'll raise a new ticket to look at the two places where we have an empty label (from an accessibility point of view).

Could you take another look at my latest commit @PatrykRudnicki ?   This one removes redundant messages and fixes a couple of issues spotted on the way.

File: conf/messages.en [hunk 7]
```diff
@@ -1284,8 +1171,6 @@ declaration.summary.transport.departure.transportCode.no-code = I don’t know
 
 declaration.summary.transport.departure.meansOfTransport.header = Transport details at the border
 declaration.summary.transport.departure.meansOfTransport.header.change = Change transport details at the border
-declaration.summary.transport.departure.meansOfTransport. =
-# this is hack about incorrectness of model ^^
```
Comment (PatrykRudnicki) [inline conf/messages.en:1288]: Are you sure this is not required?

Comment (SteveSugden) [reply conf/messages.en:1288]: Only if we don't have a meansOfTransport and we always do.

**Pull request merged**